info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -261,18 +270,46 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     datastore.update(electionEntity);
   }
 
-  // @TODO [Find incumbency.]
+  private boolean checkIncumbency(String candidateName, String position, String division) {
+    try {
+      String queryUrl = String.format("%s&ocdDivisionId=%s", REPRENTATIVE_QUERY_URL, division);
+      JsonObject representatives = queryCivicInformation(queryUrl);
+      JsonArray offices = representatives.getAsJsonArray("offices");
+      int i = 0;
+      for (JsonElement eachOffice : offices){
+        JsonObject office = (JsonObject) eachOffice;
+        String officeName = office.get("name").getAsString();
+        if (position.equals(officeName)){
```

Comment:
Instead of querying the API for each candidate, why not just query once per election and then return a map from position to current incumbant?  This function could return `Map<String, String>` that you then pass to and reference in `storeElectionContestCanddiateInDatabase()` for each candidate in an election.